src/Matrix.jl

```diff
@@ -6791,6 +6792,162 @@ end
 
 diagonal_matrix(x::NCRingElement, m::Int) = diagonal_matrix(x, m, m)
 
+###############################################################################
+#
+#   Lower triangular matrix
+#
+###############################################################################
+
+@doc Markdown.doc"""
+    lower_triangular_matrix(L::AbstractVector{T}) where {T <: RingElement}
+
+Return the $n$ by $n$ matrix whose entries on and below the main diagonal are
+the elements of `L`, and which has zeroes elsewhere.
+The value of $n$ is determined by the condition that `L` has length
+$n(n+1)/2$.
+
+An exception is thrown if there is no integer $n$ with this property.
+
+# Examples
+```jldoctest; setup = :(using AbstractAlgebra)
+julia> lower_triangular_matrix([1, 2, 3])
+[1   0]
+[2   3]
+```
+"""
+function lower_triangular_matrix(L::AbstractVector{T}) where {T <: RingElement}
+   l = length(L)
+   l == 0 && throw(ArgumentError("Input vector must be nonempty"))
+   n = Int(floor((Base.sqrt(1+8*l)-1)/2))
+   n == 0 && throw(ArgumentError("Input vector must be nonempty"))
+   l == div(n*(n+1), 2) || throw(ArgumentError("Input vector of invalid length"))
```

Comment:
@tthsqe12 Any hint how to do this without using floating point computations?

Reply:
In order to avoid floating point computations, one could call `AbstractAlgebra.sqrt` with `check` argument `false`.

Reply:
I will update the pull request.

Reply:
floating point is great
`floor(Int, sqrt(10))`
does exactly what you want?

Reply:
is there some way to macro up a constructor in comprehension form
`@TM f(i,j) for 1 <= i <= j <= n`

Reply:
Hmm, how shall we proceed, with or without floats?

Reply:
Since we want an error when the input is not a square, I would say the current version of this pr with `is_square_with_sqrt` is good.